six/modules/c++/cphd/include/cphd/Utilities.h

```diff
@@ -20,25 +20,128 @@
+ *  that defines the size of a sample in the signal block
+ *
+ *  \return Returns number of bytes: 2, 4 or 8. (0 if uninitialized)
+ */
+size_t getNumBytesPerSample(const SignalArrayFormat& signalArrayFormat);
 
+/*
+ *  \func validateFormat
+ *
+ *  \brief Validate user input format
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *
+ *  \throws except::Exception If format string is not valid binary format
+ */
+void validateFormat(const std::string& format);
+
+/*
+ *  \func getMultipleParamSizes
+ *
+ *  \brief Parse sizes of multiple params
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *
+ *  \throws except::Exception If format string is not valid binary format
+ *
+ *  \return Returns a vector of param name to param size pairs
+ */
+std::vector<std::pair<std::string,size_t> > getMultipleParamSizes(const std::string& format);
+
+/*
+ *  \func getFormatSize
+ *
+ *  \brief Return parameter size based on valid binary formats table 10.2 CPHD1.0 spec
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *
+ *  \throws except::Exception If format string is not valid binary format
+ *
+ *  \return Returns size of expected parameter
+ */
+size_t getFormatSize(const std::string& format);
+
+/*
+ *  \func isMultipleParam
+ *
+ *  \brief Checks if format is of multiple param type
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *
+ *  \return Returns true of format is multiple param type
+ */
+bool isMultipleParam(const std::string& format);
+
+/*
+ *  \func isFormatStr
+ *
+ *  \brief Checks if format is of string type
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *
+ *  \return Returns size specified by string format if true,
+    if false returns six::Init::undefined<size_t>
+ */
+size_t isFormatStr(const std::string& format);
+
+/*
+ *  \func keyValueFinder
+ *
+ *  \brief Parses format string into key value pairs. (Helper for multiple param type)
+ *
+ *  \param format A format string.
+ *   Valid binary formats are listed in CPHD 1.0 spec table 10.2, page 120
+ *  \param startPos Offset from start of format string to start parsing
+ *  \param endPos Offset from start of format string to end parsing
+ *
+ *  \throws except::Exception If format string is not valid binary format
+ *
+ *  \return std::pair<std::string, std::string> Returns pair of param name and param format
+ */
+std::pair<std::string, std::string> keyValueFinder(const std::string& format, size_t startPos, size_t endPos);
+
+/*
+ *  Ostream operators for six::CollectionInformation type
+ */
+std::ostream& operator<< (std::ostream& os, const CollectionInformation& g);
```

Comment:
Is it okay to put these here? Or should I add them to six::CollectionInformation and so on?

Reply:
I think having them in the associated files makes more sense, unless you have a strong reason to have them here.